src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -4154,4 +4154,20 @@ char *wm_vuldet_generate_msu_request(cpe *s_cpe, vu_feed dist_ver) {
     return request;
 }
 
+void wm_vuldet_normalize_architecture_NVD(char* architecture) {
```

Comment:
I wouldn’t receive a single parameter that is de-allocated within the function and then allocated again with the new string. This could be confusing for the caller as the actual value of the pointer is modified between the function.

Instead, I would make it receive the architecture string as parameter, and return a `char*` string with the translation, delegating the responsibility of freeing the memory to the caller.

Reply:
Fixed!